### PR TITLE
digdag: migrate to openjdk8

### PR DESCRIPTION
**Pull request by hyuraku**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
openjdk@8 migration (#63290)

### CODE REVIEW

File: Formula/digdag.rb [hunk 1]
```diff
@@ -11,13 +12,13 @@ class Digdag < Formula
...
 
-  depends_on java: "1.8"
+  depends_on "openjdk@8"
 
   def install
     libexec.install "digdag-#{version}.jar"
     (libexec/"bin").write_jar_script libexec/"digdag-#{version}.jar", "digdag"
     (libexec/"bin/digdag").chmod 0755
-    (bin/"digdag").write_env_script libexec/"bin/digdag", Language::Java.java_home_env("1.8")
+    (bin/"digdag").write_env_script libexec/"bin/digdag", Language::Java.java_home_env("8")
```
Comment (jonchang) [inline Formula/digdag.rb:21]: I believe this should be kept has 1.8. Does this work properly in `brew irb`?

Comment (hyuraku) [reply Formula/digdag.rb:21]: @jonchang 
Yes, `brew irb` worked as below

``` sh
$ brew irb
==> Interactive Homebrew Shell
Example commands available with: brew irb --examples

irb(main):001:0>
```

Comment (jonchang) [reply Formula/digdag.rb:21]: I just tested it myself and this needs to be changed back.

```console
% brew irb
==> Interactive Homebrew Shell
irb(main):001:0> Language::Java.java_home_env("8")
=> {:JAVA_HOME=>"$(/usr/libexec/java_home --version 8 --failfast 2>/dev/null)"}
irb(main):002:0> Language::Java.java_home_env("1.8")
=> {:JAVA_HOME=>"/usr/local/opt/openjdk@8/libexec/openjdk.jdk/Contents/Home"}
```

Comment (SMillerDev) [reply Formula/digdag.rb:21]: What about without specifying a version?

Comment (hyuraku) [reply Formula/digdag.rb]:
```bash
$ brew irb
==> Interactive Homebrew Shell

irb(main):001:0>  Language::Java.java_home_env("1.8")
=> {:JAVA_HOME=>"/usr/local/opt/openjdk@8/libexec/openjdk.jdk/Contents/Home"}
irb(main):002:0>  Language::Java.java_home_env
=> {:JAVA_HOME=>"/usr/local/opt/openjdk/libexec/openjdk.jdk/Contents/Home"}
```

**Comment by jonchang** [comment]
Thanks @hyuraku!!

**Pull request merged**